### PR TITLE
Clean up devether.c

### PR DESCRIPTION
**Pull request by dancrossnyc**
Clean up the logic to make it easier to follow.

Signed-off-by: Dan Cross <cross@gajendra.net>

### CODE REVIEW

**Comment by dancrossnyc** [comment]
FYI: Tested by booting and running Harvey.

File: sys/src/9/port/devether.c [hunk 1]
```diff
@@ -176,36 +185,40 @@ etheriq(Ether* ether, Block* bp, int fromwire)
...
-				}
-				else
-					ether->Netif.soverflows++;
-			}
-			else
+			if(f->headersonly){
 				etherrtrace(f, pkt, len);
+				continue;
+			}
+			if(fromwire && fx == 0){
```
Comment (rminnich) [inline sys/src/9/port/devether.c]: nice. I betcha this was done this way due to a compiler bug somewhere :-)

File: sys/src/9/port/devether.c [hunk 2]
```diff
@@ -245,12 +258,13 @@ etheroq(Ether* ether, Block* bp)
 		splx(s);
 	}
 
-	if(!loopback){
-		qbwrite(ether->oq, bp);
-		if(ether->transmit != nil)
-			ether->transmit(ether);
-	} else
+	if(loopback){
```
Comment (rminnich) [inline sys/src/9/port/devether.c]: geez. Wonder why they did stuff like this? Ah right ... loopback was added much later.

File: sys/src/9/port/devether.c (hunk 1 above)
Comment (dancrossnyc) [inline sys/src/9/port/devether.c]: I expect so. Barret and I talked about this once vis-a-vis Akaros, and the idea was that they may have been trying to force more efficient code generation. I think we can get around that by doing an FDO-style build.

**Pull request merged**